enclave-manager/web/src/client/KurtosisClientContext.tsx

```diff
@@ -0,0 +1,88 @@
+import { createContext, PropsWithChildren, useContext, useEffect, useState } from "react";
+import { KurtosisClient } from "./KurtosisClient";
+import { Alert, AlertDescription, AlertIcon, AlertTitle, Flex, Heading, Spinner } from "@chakra-ui/react";
+import { assertDefined, isDefined, isStringTrue, stringifyError } from "../utils";
+import { AuthenticatedKurtosisClient } from "./AuthenticatedKurtosisClient";
+import { LocalKurtosisClient } from "./LocalKurtosisClient";
+
+type KurtosisClientContextState = {
+  client: KurtosisClient | null;
+};
+
+const KurtosisClientContext = createContext<KurtosisClientContextState>({ client: null });
+
+export const KurtosisClientProvider = ({ children }: PropsWithChildren) => {
+  const [client, setClient] = useState<KurtosisClient>();
+  const [jwtToken, setJwtToken] = useState<string>();
+  const [error, setError] = useState<string>();
+
+  useEffect(() => {
+    const receiveMessage = (event: MessageEvent) => {
+      const message = event.data.message;
+      switch (message) {
+        case "jwtToken":
+          const value = event.data.value;
+          if (isDefined(value)) {
+            setJwtToken(value);
+          }
+          break;
+      }
+    };
+    window.addEventListener("message", receiveMessage);
+    return () => window.removeEventListener("message", receiveMessage);
+  }, []);
+
+  useEffect(() => {
+    const searchParams = new URLSearchParams(window.location.search);
+    const requireAuth = isStringTrue(searchParams.get("require_authentication"));
+    const requestedApiHost = searchParams.get("api_host");
+    // eslint-disable-next-line
+    const preloadedPackage = searchParams.get("package");
+    try {
+      setError(undefined);
+      if (requireAuth) {
```

Comment:
Is there any way we can move this checking out of the KurtosisClientContext? 
As a minimum I'd like to see if we can store the jwt token in a separate context provider that is tied to the authenticated client. Trying to see if we can minimize the bleeding of the need for the auth in here.
Let me know if we need to talk this over more.

Reply:
I don't think this is bleeding - this is the `KurtosisClientProvider` doing what it is responsible for - and this is the logic that switches between using a local or authenticated client.